pkg/server/v2/server.go

```diff
@@ -67,6 +65,10 @@ type Callbacks interface {
 	OnStreamRequest(int64, *discovery.DiscoveryRequest) error
 	// OnStreamResponse is called immediately prior to sending a response on a stream.
 	OnStreamResponse(int64, *discovery.DiscoveryRequest, *discovery.DiscoveryResponse)
+
+	OnStreamDeltaRequest(int64, *discovery.DeltaDiscoveryRequest) error
+	OnStreamDeltaResponse(int64, *discovery.DeltaDiscoveryRequest, *discovery.DeltaDiscoveryResponse)
```

Comment:
I've tried implementing delta xds and I guess this one is not correct. In delta XDS the response cannot necessarily be attributed to particular request, hence IMO - this one should be without the "request" part

Reply:
So you can definitely correlate the initial response to the initial request.  Is your point that you might receive multiple requests on the same stream and therefore can't correctly attribute a response to a specific the request?  If so, I think that makes sense especially since the subsequent requests may simply add (or remove) a few new resource subscriptions therefore attributing a response that is not called out in those requests would be a bit odd. @alecholmez, do you concur?

Reply:
Yes, the sotw api has both, because to any request there is a corresponding response. For delta api requests are for modification of internal state and that state is the trigger for doing responses, hence you cannot attribute response to particular request.
That being said - if we drop the request here - we would not be able to easily find out what node this response belongs to.

Reply:
I do concur that makes sense. I guess I could pull the request object out of that callback. Is the general consensus to remove that? I'd rather keep it in as there is some useful information in that request object, though you are both correct, a strict correlation between the initial request and following responses wouldn't make sense.